src/main/java/org/fiware/iam/til/TrustedIssuersListAdapter.java

```diff
@@ -0,0 +1,51 @@
+package org.fiware.iam.til;
+
+import io.micronaut.http.HttpResponse;
+import jakarta.inject.Singleton;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.iam.til.api.IssuerApiClient;
+import org.fiware.iam.til.model.ClaimVO;
+import org.fiware.iam.til.model.CredentialsVO;
+import org.fiware.iam.til.model.TrustedIssuerVO;
+
+import java.util.List;
+import java.util.Optional;
+
+@Singleton
+@RequiredArgsConstructor
+@Slf4j
+public class TrustedIssuersListAdapter {
+
+    private final IssuerApiClient apiClient;
+    private final TrustedIssuerConfigProvider trustedIssuerConfigProvider;
+
+    public void allowIssuer(String serviceDid) {
+        CredentialsVO credentialToBeAdded = trustedIssuerConfigProvider.createCredentialConfigForTargetService();
+        try {
+            Optional<TrustedIssuerVO> issuer = getIssuer(serviceDid);
+            if (issuer.isPresent()) {
+                TrustedIssuerVO updatedIssuer = issuer.get().addCredentialsItem(credentialToBeAdded);
+                apiClient.updateIssuer(serviceDid, updatedIssuer);
+            } else {
+                apiClient.createTrustedIssuer(new TrustedIssuerVO().did(serviceDid).addCredentialsItem(credentialToBeAdded));
+            }
```

Comment:
Why do we need a catch all here? And why don't we bubble the exception? Wouldn't it make sense that the sender(e.g. TMForum) gets informed(e.g. 502 or something) and can retry at some point? With that solution, a user could purchase access to a service, but never actually get the access, without anyone knowing about.

Reply:
Do we want the tm forum service to take care of that, or shall we retry in this service?
What is the expected behavior? "User can buy a service but can only buy it if all downstream components are fine." feels a bulky (unless we span a transaction over all components and roll back)

Reply:
The TMForum notifications are decoupled from the normal crud operations and will retry. However, this service could do this also. Most importantly, it needs to log issues properly. So i would be fine if it does not retry(its blocking anyways, so it would require some restructuring)

Reply:
So extend the logging and have a second version later with eg some kind of external msg queue where the notifications are placed and pulled for execution?

Reply:
Logging and a 502 would be fine for now. Then the tmforum handles the rest